pwiz_tools/Skyline/Model/Lib/LibraryFiles.cs

```diff
@@ -34,17 +34,20 @@ namespace pwiz.Skyline.Model.Lib
     /// </summary>
     public class LibraryFiles : IReadOnlyList<string>
     {
-        public static LibraryFiles EMPTY = new LibraryFiles(Array.Empty<string>());
+        public static LibraryFiles EMPTY = new LibraryFiles(Array.Empty<string>(), Array.Empty<WorkflowType>());
 
         private Dictionary<MsDataFileUri, int> _msDataFileUriLookup = new Dictionary<MsDataFileUri, int>();
         private ImmutableList<string> _baseNames;
-        public LibraryFiles(IEnumerable<string> sourceFilePaths)
+        public LibraryFiles(IEnumerable<string> sourceFilePaths, IEnumerable<WorkflowType> workflowTypes)
         {
             FilePaths = ImmutableList.ValueOf(sourceFilePaths);
+            WorkflowTypes = ImmutableList.ValueOf(workflowTypes);
             _baseNames = ImmutableList.ValueOf(FilePaths.Select(GetBaseName));
         }
 
         public ImmutableList<string> FilePaths { get; }
+        public ImmutableList<WorkflowType> WorkflowTypes { get; }
```

Comment:
I would rather keep the class "LibraryFiles" simple so that its only data member is a list of file paths.

It would be better if you could use "Library.LibraryDetails" to answer the question of which WorkflowTypes there are, instead of "Library.LibraryFiles".

One problem with changing "ImportPeptideSearchDlg.GetAreLibrarySpectraDia" to use "LibraryDetails", though, is that it is incredibly slow for very large .blib files because it has to do a JOIN in order to answer the question of how many "best" spectra each file has.
However, since we are changing the .slc file format anyway, we could also change "BiblioLiteSpectrumInfo" so that it knows which file has the best spectrum, and then we could remove all of the SQL statements from BiblioSpecLiteLibrary.GetDataFileDetails.